lib/cores.js

```diff
@@ -108,36 +109,45 @@ async function getConnectedCoreIPsAsync(coreIPList, coreConnectionCount) {
   }
 }
 
-async function coreRequestAsync(options, coreIP, retryCount = 3) {
+async function coreRequestAsync(options, coreIP, retryCount = 3, timeout = 500) {
   options.headers['X-Node-Version'] = version
   options.uri = `http://${coreIP}${options.uriPath}`
 
   let response
   if (retryCount <= 0) {
-    response = await rp(options)
+    try {
+      response = await rp(options)
+    } catch (error) {
+      if (error.statusCode === 402 || error.status === 402) return error.response
+      else throw error
+    }
   } else {
     await retry(
       async bail => {
         try {
           response = await rp(options)
         } catch (error) {
-          // If no response was received or there is a status code >= 500, then we should retry the call, throw an error
-          if (!error.statusCode || error.statusCode >= 500) throw error
+          // If no response was received or there is a status code >= 500 or payment is still pending/held,
+          // then we should retry the call, throw an error
+          if (!error.statusCode || error.statusCode >= 500 || error.statusCode === 402) throw error
           // errors like 409 Conflict or 400 Bad Request are not retried because the request is bad and will never succeed
           bail(error)
         }
       },
       {
         retries: retryCount, // The maximum amount of times to retry the operation. Default is 3
         factor: 1, // The exponential factor to use. Default is 2
-        minTimeout: 500, // The number of milliseconds before starting the first retry. Default is 200
+        minTimeout: timeout, // The number of milliseconds before starting the first retry. Default is 200
         randomize: true,
         onRetry: error => {
-          logger.warn(
-            `Core request : ${error.statusCode || 'no response'} : Core ${coreIP} : Request ${
-              options.uri
-            } - ${JSON.stringify(options)} : ${error.message} : retrying`
-          )
+          if (error.statusCode === 402)
+            logger.warn(`Core request : 402: Payment Required. Core ${coreIP} : Request ${options.uri}. Retrying`)
+          else
+            logger.warn(
+              `Core request : ${error.statusCode || 'no response'} : Core ${coreIP} : Request ${
+                options.uri
+              } - ${JSON.stringify(options, null, 2)} : ${error.message} : retrying`
+            )
```

Comment:
Cleanup of error messages. Also for 402 we don't need the whole response since we expect this to happen once or twice.